kotest-runner/kotest-runner-junit5/src/jvmMain/kotlin/io/kotest/runner/junit/platform/gradle/GradleClassMethodRegexTestFilter.kt

```diff
@@ -18,41 +18,58 @@ class GradleClassMethodRegexTestFilter(private val patterns: List<String>) : Tes
       }
    }
 
-   fun match(pattern: String, descriptor: Descriptor): Boolean {
-      val (pck, classname, path) = GradleTestPattern.parse(pattern)
+   private fun match(pattern: String, descriptor: Descriptor): Boolean {
+      val (prefixWildcard, pck, classname, path) = GradleTestPattern.parse(pattern)
       return when (descriptor) {
-         is Descriptor.TestDescriptor -> descriptor.parts().take(path.size) == path
+         is Descriptor.TestDescriptor -> when (path) {
+           null -> false
+           else -> descriptor.path(false).value.startsWith(path)
+         }
+
          is Descriptor.SpecDescriptor -> when {
+            pck != null && classname != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains("$pck.$classname") ?: false
             pck != null && classname != null -> descriptor.kclass.qualifiedName == "$pck.$classname"
+            pck != null && prefixWildcard -> descriptor.kclass.qualifiedName?.contains(pck) ?: true
             pck != null -> descriptor.kclass.qualifiedName?.startsWith(pck) ?: true
+            classname != null && prefixWildcard -> descriptor.kclass.simpleName?.contains(classname) ?: false
             classname != null -> descriptor.kclass.simpleName == classname
             else -> true
```

Comment:
```suggestion
         is Descriptor.SpecDescriptor -> when {
            pck == null && classname == null -> true
            pck == null && prefixWildcard -> descriptor.kclass.simpleName?.contains(classname) ?: false
            pck == null -> descriptor.kclass.simpleName == classname
            prefixWildcard -> descriptor.kclass.simpleName?.contains(classname) ?: false
            else -> descriptor.kclass.simpleName == classname
```

I was thinking of maybe inverting the when conditions, would lead to less code although I don't think it's much cleaner hahah

I didn't check if they're the same thing either

Reply:
_I didn't check if they're the same thing either_
well this works so I suggest we leave it ;)